### DISCUSSION TITLE
Another pure slint workaround for reacting to property changes

### DISCUSSION THREAD

**Discussion by maurges** [slint-ui/slint#4717] [General]
Similar to #4324, but this time abusing pure callbacks.

```
component ChangeObserver {
    in property <length> value;
    // first argument is new value; must return second argument
    pure callback changed(length, float) -> float;

    width: 0; height: 0; visible: false;

    opacity: changed(value, 1);
}
```
This uses the fact that the callback is going to be called on every value (modulo waiting for frame shenanigans I guess).

I also found that slint compiler is smarter than I thought and will remove a dummy custom property that is never read, so I had to use an inbuilt one. I never change opacity in my code, especially dynamically, but if you do, you can pick something else.

**Comment by ogoffart** [comment]
Thanks, this is similar to the workaround from https://github.com/slint-ui/slint/issues/112#issuecomment-1065866865 but updated and generalized.
I added a link to this discussion from the issue.
